airbyte-integrations/connectors/source-cart/source_cart/streams.py

```diff
@@ -0,0 +1,141 @@
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+import urllib.parse
+from abc import ABC
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class CartStream(HttpStream, ABC):
+    primary_key = "id"
+
+    def __init__(self, start_date: str, store_name: str, **kwargs):
+        self._start_date = start_date
+        self.store_name = store_name
+        super().__init__(**kwargs)
+
+    @property
+    def url_base(self) -> str:
+        return f"https://{self.store_name}/api/v1/"
+
+    def backoff_time(self, response: requests.Response) -> Optional[float]:
+        """
+        We dont need to check the response.status_code == 429 since this header exists only in this case.
+        """
+        retry_after = response.headers.get("Retry-After")
+        if retry_after:
+            return float(retry_after)
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        response_json = response.json()
+        if response_json.get("next_page"):
+            next_query_string = urllib.parse.urlsplit(response_json.get("next_page")).query
+            params = dict(urllib.parse.parse_qsl(next_query_string))
+            return params
+
+    def request_headers(self, **kwargs) -> Mapping[str, Any]:
+        return {"Cache-Control": "no-cache", "Content-Type": "application/json"}
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        result = response_json.get(self.data_field, [])
+        yield from result
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = {"count": 100, "sort": "updated_at"}
+        if next_page_token:
+            params.update(next_page_token)
+        return params
+
+
+class IncrementalCartStream(CartStream, ABC):
+
+    state_checkpoint_interval = 1000
+    cursor_field = "updated_at"
+
+    def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        cursor_value = stream_state.get(self.cursor_field) or self._start_date
+        params[self.cursor_field] = f"gt:{max(cursor_value, self._start_date)}"
```

Comment:
this is not ok to use unrelated vars here as the name of request parameter

Reply:
this parameter is date filter (date > state_value) to support incremental